src/clojure/clj_yaml/core.clj

```diff
@@ -149,22 +149,34 @@
   (.dump ^Yaml (apply make-yaml opts)
          (encode data)))
 
+(defn- load-str [^Yaml yaml ^String input load-all? keywords]
+  (if load-all?
+    (map #(decode % keywords) (.loadAll yaml input))
+    (decode (.load yaml input) keywords)))
+
 (defn parse-string
-  [^String string & {:keys [unsafe mark keywords max-aliases-for-collections allow-recursive-keys allow-duplicate-keys] :or {keywords true}}]
-  (decode (.load (make-yaml :unsafe unsafe
-                            :mark mark
-                            :max-aliases-for-collections max-aliases-for-collections
-                            :allow-recursive-keys allow-recursive-keys
-                            :allow-duplicate-keys allow-duplicate-keys)
-                 string) keywords))
+  [^String string & {:keys [unsafe mark keywords max-aliases-for-collections allow-recursive-keys allow-duplicate-keys load-all?] :or {keywords true}}]
+  (load-str (make-yaml :unsafe unsafe
+                       :mark mark
+                       :max-aliases-for-collections max-aliases-for-collections
+                       :allow-recursive-keys allow-recursive-keys
+                       :allow-duplicate-keys allow-duplicate-keys)
+            string
+            load-all? keywords))
 
 ;; From https://github.com/metosin/muuntaja/pull/94/files
 (defn generate-stream
   "Dump the content of data as yaml into writer."
   [writer data & opts]
   (.dump ^Yaml (apply make-yaml opts) (encode data) writer))
 
+(defn- load-stream [^Yaml yaml ^java.io.Reader input load-all? keywords]
```

Comment:
This looks a lot like `load-str`, maybe they can be implemented in terms of each other?
By creating a `Reader` from a `String` via the `StringReader` you might just be able to eliminate `load-str` ?

Reply:
Fair play, patch for this in f2d68bb